src/main/java/org/ballerinalang/command/cmd/ListCommand.java

```diff
@@ -91,31 +97,71 @@ public void setParentCmdParser(CommandLine parentCmdParser) {
-                            + " " +  ToolUtil.getType(version) + " version " + version);
-                    installedVersions.add(version);
+            for (Channel channel : channels) {
+                for (Distribution distribution : channel.getDistributions()) {
+                    for (int i = 0; i < listOfFiles.length; i++) {
+                        if (listOfFiles[i].isDirectory()) {
+                            String version = listOfFiles[i].getName().split("-")[1];
+                            if (version.equals(distribution.getVersion())) {
+                                outStream.println(markVersion(currentBallerinaVersion, version)
+                                        + " " + ToolUtil.getType(version));
+                                installedVersions.add(version);
+                            }
+                        }
+                    }
                 }
             }
-            outStream.println("\nDistributions available remotely:");
-            List<Channel> channels = ToolUtil.getDistributions();
+            FileWriter writer = null;
+            try {
+                writer = new FileWriter("localDists.json");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            for(String str: installedVersions) {
+                writer.write(str + System.lineSeparator());
+            }
+            writer.close();
+            outStream.println("\nDistributions available remotely:\n");
             for (Channel channel : channels) {
                 outStream.println("\n" + channel.getName() + "\n");
                 for (Distribution distribution : channel.getDistributions()) {
-                    outStream.println("  [" + distribution.getVersion() + "] " + distribution.getName());
+                    outStream.println(markVersion(currentBallerinaVersion, distribution.getVersion())
+                            + " " + distribution.getName());
                 }
             }
         } catch (CommandException e) {
             ErrorUtil.printLauncherException(e, outStream);
+            outStream.println("Distributions available locally: \n");
+            BufferedReader reader;
+            try {
+                reader = new BufferedReader(new FileReader("localDists.json"));
+                String line = reader.readLine();
+                while (line != null) {
+                    try {
+                        line = reader.readLine();
+                    } catch (IOException ioException) {
+                        ioException.printStackTrace();
+                    }
+                    outStream.println(markVersion(currentBallerinaVersion, line)
+                            + "  " + ToolUtil.getType(line));
+                }
+                reader.close();
+            } catch (FileNotFoundException fileNotFoundException) {
+                fileNotFoundException.printStackTrace();
```

Comment:
@parkavi11 Usually we do not print the exception like this we should handle it and throw Command exceptions as we have done in other places